apps/challenges/views.py

```diff
@@ -677,20 +681,78 @@ def create_challenge_using_zip_file(request, challenge_host_team_pk):
     """
     Creates a challenge using a zip file.
     """
+    if request.data.get("is_template_challenge"):
+        is_template_challenge = True
+    else:
+        is_template_challenge = False
+
     challenge_host_team = get_challenge_host_team_model(challenge_host_team_pk)
 
-    serializer = ChallengeConfigSerializer(
-        data=request.data, context={"request": request}
-    )
+    # All files download and extract location.
+    BASE_LOCATION = tempfile.mkdtemp()
+
+    if is_template_challenge:
+        template_id = request.data.get("templateId")
+        try:
+            template = ChallengeTemplate.objects.get(
```

Comment:
can you create a function like `get_challenge_model` for the challenge template get too. And use that here.